src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,211 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+import duke.file.FileManager;
+
+/**
+ * Stores and add to the list of tasks.
+ */
+public class TaskManager implements TaskAction {
+    private static final String BY_LITERAL = " /by ";
+    private static final String AT_LITERAL = " /at ";
+    private static final int MAX_INPUT_PARAMS = 2;
+
+    private static final String ERROR_NOT_INTEGER = "Sorry but parameter entered is not a integer\n";
+    private static final String ERROR_NOT_IN_RANGE = "Sorry but parameter entered is not within range of list\n";
+    private static final String ERROR_TASK_TYPE_NOT_FOUND = "Task type is not found\n";
+    private static final String ERROR_LIST_EMPTY = "No task in your list. Add some!\n";
+
+    private final ArrayList<Task> tasks;
+
+    public TaskManager() {
+        tasks = new ArrayList<>();
+    }
+
+    public TaskManager(ArrayList<Task> previousData) {
+        tasks = previousData;
+    }
+
+    /**
+     * Adds a task to the user's list of tasks.
+     * Prints to user that task has been added.
+     *
+     * @param taskType represents the type of task to store.
+     * @param inputText text of the user input to store.
+     */
+    @Override
+    public void addTask(TaskType taskType, String inputText) {
+        Task task;
+        String[] taskParameters;
+
+        try {
+            switch (taskType) {
+            case TODO:
+                task = new Todo(inputText);
+                break;
+            case DEADLINE:
+                taskParameters = splitItem(TaskType.DEADLINE, inputText);
+                task = new Deadline(taskParameters);
+                break;
+            case EVENT:
+                taskParameters = splitItem(TaskType.EVENT, inputText);
+                task = new Event(taskParameters);
+                break;
+            default:
+                throw new IllegalStateException();
+            }
+            addTaskToList(task);
+            printAddTaskSuccessful(task);
+            FileManager.writeToFile(FileManager.FILE_PATH, tasks);
+        } catch (MissingTaskLiteralException e) {
+            printMissingLiteral(e.getMessage());
+        } catch (IllegalStateException e) {
+            printTaskNotFound();
+        }
+    }
+
+    public void addTask(Task task) {
+        addTaskToList(task);
+        printAddTaskSuccessful(task);
+        FileManager.writeToFile(FileManager.FILE_PATH, tasks);
+    }
+
+    /**
+     * Prints the entire list of user's tasks.
+     */
+    @Override
+    public void printAllTasks() {
+        if (tasks.size() == 0) {
```

Comment:
Try to reduce the use of magic numbers. Maybe try to declare it in a constant variable?

Reply:
Yep will do!